Gems/ROS2/Code/Source/Sensor/SensorHelpers.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ROS2/Sensor/Events/PhysicsBasedSource.h>
+#include <ROS2/Sensor/Events/TickBasedSource.h>
+#include <ROS2/Sensor/ROS2SensorComponentBase.h>
+#include <ROS2/Sensor/SensorHelper.h>
+
+namespace ROS2
+{
+
+    AZStd::vector<AZ::EntityComponentIdPair> GetSensorsForEntity(const AZ::EntityId& entityId)
+    {
+        AZStd::vector<AZ::EntityComponentIdPair> sensors;
+        AZ::Entity* entity = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->FindEntity(entityId);
+        AZ_Assert(entity, "Entity not found");
+        if (entity)
+        {
+            auto components = entity->GetComponents();
+            for (auto component : components)
+            {
+                if (IsComponentROS2Sensor(component))
+                {
+                    sensors.push_back(AZ::EntityComponentIdPair(entityId, component->GetId()));
+                }
+            }
+        }
+        return sensors;
+    }
+
+    AZStd::vector<AZ::EntityComponentIdPair> GetSensorsOfType(const AZ::EntityId& entityId, const AZ::Uuid& sensorType)
+    {
+        AZStd::vector<AZ::EntityComponentIdPair> sensors;
+        AZ::Entity* entity = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->FindEntity(entityId);
+        AZ_Assert(entity, "Entity not found");
+        if (entity)
+        {
+            auto components = entity->GetComponents();
+            for (auto component : components)
+            {
+                AZ_Assert(component, "Component not found");
+                const auto ptr = component->RTTI_IsTypeOf(sensorType);
+                if (ptr)
```

Comment:
```suggestion
                if (component->RTTI_IsTypeOf(sensorType))
```
RTTI_IsTypeOf returns a bool, not a ptr